pkgs/by-name/fr/freedownloadmanager/package.nix

```
@@ -0,0 +1,124 @@
+  };
+
+  unpackPhase = "dpkg-deb -x $src .";
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    patchelf
+    wrapGAppsHook3
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  buildInputs =
+    [
+      # Runtime libraries identified as needed by FDM or its bundled Qt libs
+      libdrm
+      libpqxx # Bundled Qt SQL plugin dependency (PostgreSQL)
+      unixODBC # Bundled Qt SQL plugin dependency (ODBC)
+      libpulseaudio # For QtMultimedia
+      # XCB utils needed by Qt platform plugin
+      xcbutilcursor
+      xcbutilwm
+      xcbutilimage
+      xcbutilkeysyms
+      xcbutilrenderutil
+      (lib.getLib udev) # For device access, if needed beyond what hooks provide
+    ]
+    ++ (with gst_all_1; [
+      gstreamer
+      gst-libav
+      gst-plugins-base
+      gst-plugins-good
+      gst-plugins-bad
+      gst-plugins-ugly
+    ]);
+
+  # Note: autoPatchelfIgnoreMissing = [ ... ] failed during testing (NixOS unstable 24.11, 2025-04-05).
+  # So I used a manual patchelf in installPhase as a workaround. You can see my commit message for more information.
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/opt/${pname} $out/share/pixmaps
+
+    cp -r opt/freedownloadmanager/* $out/opt/${pname}/
+    cp -r usr/share/* $out/share/
+
+    ln -s $out/opt/${pname}/fdm $out/bin/${pname}
+
+    ln -s $out/opt/${pname}/icon.png $out/share/pixmaps/${pname}.png
+
+    substituteInPlace $out/share/applications/freedownloadmanager.desktop \
+      --replace "Exec=/opt/freedownloadmanager/fdm" "Exec=${pname}" \
+      --replace "Icon=/opt/freedownloadmanager/icon.png" "Icon=${pname}"
+
+    echo "Attempting to manually remove optional SQL dependencies..."
+    local sql_plugin_dir="$out/opt/${pname}/plugins/sqldrivers"
+
+    if [ -f "$sql_plugin_dir/libqsqlmimer.so" ]; then
+      patchelf --remove-needed libmimerapi.so "$sql_plugin_dir/libqsqlmimer.so" \
+        || echo "Warning: Failed to remove libmimerapi.so dependency from libqsqlmimer.so (or already removed/not found)."
+    else
+       echo "Info: libqsqlmimer.so not found, skipping patchelf."
+    fi
+
+    if [ -f "$sql_plugin_dir/libqsqlmysql.so" ]; then
+      patchelf --remove-needed libmysqlclient.so.21 "$sql_plugin_dir/libqsqlmysql.so" \
+        || echo "Warning: Failed to remove libmysqlclient.so.21 dependency from libqsqlmysql.so (or already removed/not found)."
+    else
+      echo "Info: libqsqlmysql.so not found, skipping patchelf."
+    fi
+    echo "Manual SQL dependency removal finished."
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Download manager supporting many protocols";
+    homepage = "https://www.freedownloadmanager.org";
+    license = licenses.unfree;
+    platforms = [ "x86_64-linux" ];
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    maintainers = with maintainers; [ maintainers.scriptod ];
+    mainProgram = "fdm";
+  };
```

Quick formatting nitpick:
```suggestion
  meta = {
    description = "Download manager supporting many protocols";
    homepage = "https://www.freedownloadmanager.org";
    license = lib.licenses.unfree;
    platforms = [ "x86_64-linux" ];
    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
    maintainers = with lib.maintainers; [ scriptod ];
    mainProgram = "fdm";
  };
```